scripts/benchmarks/output_markdown.py

```diff
@@ -1,220 +1,410 @@
 # SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
 
-import collections, re
+import collections
 from benches.result import Result
-from options import options
+from options import options, MarkdownSize
 import math
+import ast
 
 class OutputLine:
     def __init__(self, name):
         self.label = name
         self.diff = None
         self.bars = None
         self.row = ""
+        self.suite = "Unknown"
+        self.explicit_group = ""
 
     def __str__(self):
         return f"(Label:{self.label}, diff:{self.diff})"
 
     def __repr__(self):
         return self.__str__()
 
-# Function to generate the markdown collapsible sections for each variant
-def generate_markdown_details(results: list[Result]):
-    markdown_sections = []
+# The number of the required columns in the markdown table,
+# independent of the chart_data content.
+# Required columns:
+# - benchmark_name
+#
+# optional +1: relative performance
+num_info_columns = 1
+
+# Number of columns required for relative performance change calculation.
+# In case of multiple provided saved baselines to compare, the relative
+# performance is not calculated, since the base (hopefully) usage case 
+# for this script would be comparing the performance of PR with the main branch
+num_baselines_required_for_rel_change = 2
+
+# Maximum number of characters that is allowed in request validation
+# for posting comments in GitHub PRs
+max_markdown_size = 65536
+
+
+def is_relative_perf_comparison_to_be_performed(chart_data: 
+                                                dict[str, list[Result]], 
+                                                baseline_name: str):
+    return (len(chart_data) == num_baselines_required_for_rel_change) and \
+            (baseline_name in chart_data.keys())
+    
+
+def get_chart_markdown_header(chart_data: dict[str, list[Result]], 
+                              baseline_name: str):
+    summary_header = ''
+    final_num_columns = num_info_columns
+
+    if is_relative_perf_comparison_to_be_performed(chart_data, baseline_name):
+        summary_header = "| Benchmark | " + " | ".join(chart_data.keys()) + \
+                        " | Change |\n"
+        final_num_columns += 1
+    else:
+        summary_header = "| Benchmark | " + " | ".join(chart_data.keys()) + \
+                        " |\n"
+
+    summary_header += "|---" * (len(chart_data) + final_num_columns) + "|\n"
+
+    return summary_header
+
+
+def get_improved_regressed_summary(is_improved: bool, rows_count: int):
+    title = "Improved"
+    if not is_improved:
+        title = "Regressed"
+
+    summary = (
+            "\n<details>\n"
+            "<summary>\n"        
+            f"{title} {rows_count} "
+            f"(threshold {options.epsilon*100:.2f}%)\n" 
+            "</summary>\n\n"
+            )
+
+    return summary
+
+
+def get_relative_perf_summary(group_size: int, diffs_product: int, 
+                              root_for_geometric_mean: int, group_name: str):
```

Comment:
geomean is complicated to calculate. I'd just replace the : xx % change with (improved X, regressed Y)